compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/AbstractParserErrorHandler.java

```diff
@@ -112,13 +112,14 @@ public Solution recover(ParserRuleContext currentCtx, STToken nextToken, Object.
         }
 
         // Fail safe. This means we can't find a path to recover.
+        // TODO: Enable fail safe assert
+//        assert false : "fail safe reached";
```

Comment:
@rdulmina Apparently, we do have some parser tests reaching the fail-safe in the **master**. Created an issue to enable this assertion.

Reply:
I think we can't have an assert here because we reach here for `bestMatch.matches = 0` as well which can be a correct execution.

Reply:
Else lets have an assert with checking `itterCount == ITTER_LIMIT`

Reply:
my bad. +1

Reply:
Some LS tests are hitting this assertion. Hence re-opened the issue #29947